src/Control/Monad/Static.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE StaticPointers #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Control.Monad.Static where
+
+import Control.Applicative.Static
+import Control.Distributed.Closure
+import Data.Functor.Static
+import Data.Typeable (Typeable)
+
+-- | Instances of 'StaticBind' should satisfy the following laws (writing
+-- 'staticMap', 'staticApply', 'staticBind' as infix @('<$>')@, @('<*>')@, @(>>=)@,
+-- respectively):
+--
+-- @
+-- (m >>= f) >>= g = m >>= static (.) ``cap`` (staticFlippedBind g) `cap` f
```

Comment:
Right. You can for any closed type `m`. It's a bit fishy as it doesn't quite allow to express monad-transformer like things (their laws seem to only hold for static monad arguments).

It may be fine. But it may also suggest that the primitive method is something like
```haskell
   staticKleisliExtension :: (Typeable a, Typeable b) => Closure (a -> m b) -> Closure (m a -> m b)
```
From which `staticBind` has an obvious default implementation (`staticBind = flip (unclosure . staticKleisliExtension)`).

I don't know how much this matters, and how much I'm overthinking this :-p .